token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -122,6 +122,15 @@ fn process_configure_account(
     let mint = StateWithExtensions::<Mint>::unpack(mint_data)?;
     let confidential_transfer_mint = mint.get_extension::<ConfidentialTransferMint>()?;
 
+    // If a `ConfidentialTransferAccount` extension is already configured, the instruction fails to
+    // prevent overwriting existing data.
+    if token_account
+        .get_extension::<ConfidentialTransferAccount>()
+        .is_ok()
+    {
+        return Err(TokenError::ExtensionAlreadyInitialized.into());
+    }
+
     // Note: The caller is expected to use the `Reallocate` instruction to ensure there is
     // sufficient room in their token account for the new `ConfidentialTransferAccount` extension
     let mut confidential_transfer_account =
```

Comment:
nit: what do you think about adding a flag to `init_extension` to fail if the extension already exists? because otherwise this behavior will be a footgun for any extension that's initialized after the account

Reply:
Yes, I was actually just about to ask you about that. `get_extension` also makes unnecessary copies if we just care about whether an extension is initialized or not. I think adding a flag to `init_extension` is a better and more natural way.

Reply:
Ok great! I hope there won't be too much churn because of the change :crossed_fingers: